lib/puppet/provider/user/directoryservice.rb

```diff
@@ -401,6 +401,13 @@ def iterations=(value)
   # we have to treat the ds cache just like you would in the password=
   # method.
   def salt=(value)
+    if (Puppet::Util::Package.versioncmp(self.class.get_os_version, '10.15') >= 0)
+      if value.length != 64
+        raise Puppet::Error, "OS X versions 10.15 and higher require the salt to be 32-bytes. Since Puppet's user 
```

Comment:
This is nitpicky, but versions of Mac >= 10.12 are referred to as macOS, not OS X.

Reply:
Oh good catch, I didn't know that. I'll go ahead and fix that. Thank you!